File: HTO/HTO.Web/Server/Models/ChatMessage.cs
using HTO.Web.Server.Enums;
using System;

namespace HTO.Web.Server.Models
{
    public class ChatMessage
    {
        public AppTypes AppType { get; set; }
        public string Message { get; set; }
        public DateTime? ReceivedDateTime { get; set; }
        public DateTime SendDateTime { get; set; }
		public string Token { get; set; }
	}
}

File: HTO/HTO.Web/Client/Core/Models/chatMessage.js

(function (hto) {
	"use strict";

	function ChatMessage() {
		/// <summary>
		/// Represents a message sent from the mobile application.
		/// </summary>

	    this.appType = null;
	    this.message = null;
	    this.receivedDateTime = null;
	    this.sendDateTime = null;
	    this.token = null;
	}

	hto.models.ChatMessage = ChatMessage;

}(hto));
